### ISSUE TITLE
Support TLS offload

### ISSUE THREAD

**Issue by jstangroome** [logstash-plugins/logstash-input-lumberjack#60]
I run several servers with logstash-input-lumberjack behind an AWS ELB.

At the moment I have the ELB configured for TCP=>TCP proxying of the lumberjack port (5043) and I have the TLS private key installed on each logstash server.

For centralised certificate management (and maybe a little CPU offload too) I'd like to be able to configure the ELB for TLS=>TCP proxying. That is, I want to install the certificate and private key on the ELB and then have the ELB proxy to Logstash without TLS so that I do not need to manage the certificates on each Logstash server.

**logstash-forwarder clients will still be connecting to the ELB over TLS**

Can we enable an option for the lumberjack input to allow the TLS part of the protocol to be terminated by a downstream proxy? Maybe do this by adding support for the [PROXY protocol](http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt) to allow the ELB (or other proxies) to communicate to the lumberjack input that TLS has been offloaded.

**Comment by ph** [comment]
@jstangroome this is only a matter of supporting plain text connection? No?

**Comment by jstangroome** [comment]
@ph Yes, supporting plain text connection. I just wanted to qualify the scenario was not to encourage plain-text log shipping end-to-end.

**Comment by ph** [comment]
@jstangroome Agree I think #56 would solve your problem?

**Comment by jstangroome** [comment]
@ph yes. I have something similar to #56 in a private build that has been working but would love to return to the mainline.
